### PR TITLE
Remove trigger_typing / update the typing context manager

### PR DESCRIPTION
**Pull request by NextChai**
## Summary

This PR removes the coroutine `Messageable.trigger_typing` and replaces it with only `Messageable.typing`. To do so, the `Typing` context manager has been updated to allow for the user to await it. This pr, from a user's perspective, simplifies sending a typing notification to a destination.

Previously:
```python
await ctx.trigger_typing()
# do some computational magic
await ctx.send('Done!')

async with ctx.typing():
    await asyncio.sleep(20)
    await ctx.send('Done')
```

Updated:
```python
await ctx.typing()
# do some computational magic
await ctx.send('Done!')

async with ctx.typing():
    await asycio.sleep(20)
    await ctx.send('Done!')
```

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/context_managers.py [hunk 1]
```diff
@@ -53,21 +53,35 @@ class Typing:
...
+
+        self.channel = channel = await self.messageable._get_channel()
+        return channel
+
+    async def wrapped_typer(self) -> None:
+        channel = await self._get_channel()
+        await channel._state.http.send_typing(channel.id)
+
+    def __call__(self) -> Coroutine[Any, Any, None]:
+        return self.wrapped_typer()
```
Comment (Gobot1234) [inline discord/context_managers.py:70]: This isn't required unless you want to support
```py
async with ctx.typing()(): ...
```
right?

Comment (NextChai) [reply discord/context_managers.py:70]: oop! yes that's correct, committing to fix this now

File: discord/abc.py [hunk 2]
```diff
@@ -1543,32 +1543,30 @@ async def send(
...
-        channel = await self._get_channel()
-        await self._state.http.send_typing(channel.id)
-
     def typing(self) -> Typing:
-        """Returns an asynchronous context manager that allows you to type for an indefinite period of time.
-
-        This is useful for denoting long computations in your bot.
+        """Returns an asynchronous context manager that allows you to send a typing indicator to
+        the destination for an indefinite period of time, or 10 seconds if the context manager
+        is ``awaited``.
```
Comment (Rapptz) [inline discord/abc.py:1549]:
```suggestion
        is called using ``await``.
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
Needs documentation in migrating.rst

File: docs/migrating.rst [hunk 3]
```diff
@@ -1149,6 +1149,10 @@ The following have been removed:
 
     - Consider using the newer documented :func:`on_socket_event_type` event instead.
 
+- ``abc.Messageable.trigger_typing``
+
+    - Use :meth:`abc.Messageable.typing` instead.
```
Comment (Rapptz) [inline docs/migrating.rst:1154]:
```suggestion
    - Use :meth:`abc.Messageable.typing` with ``await`` instead.
```

**Review by Rapptz — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**